tokio-net/src/process/mod.rs

```diff
@@ -577,21 +570,24 @@ impl Command {
     /// async fn run_ls() -> std::process::ExitStatus {
     ///     Command::new("ls")
     ///         .status()
-    ///         .expect("ls command failed to start")
     ///         .await
     ///         .expect("ls command failed to run")
     /// }
-    pub fn status(&mut self) -> io::Result<StatusAsync> {
-        self.spawn().map(|mut child| {
+    pub fn status(&mut self) -> impl Future<Output = io::Result<ExitStatus>> {
```

Comment:
Had to manually use `impl Future` here to ensure that the result's lifetime isn't tied to the self borrow here (we only need to borrow to spawn, but not to run the future to completion).

However, I'm not sure if this declaration is enough or if we need to specify `impl Future<Output = ...> + Send + Sync` for backwards compatibility?

Reply:
Without Send / Sync is fine I think. 

What error did you get with async fn? Is there a rustc bug that is related?

Reply:
One of the existing tests now started failing due to a dropped temporary:

```rust
// Original code, constructs a command and spawns it
let child = cat().status();

wait_with_output(child).await?
```

Thus, before you'd be able to write code like the following because the `status` future does not borrow the command struct.

```rust
let mut cmd = Command::new(...);
let mut status = cmd.status();
drop(cmd);

status.await?
```

Reply:
It's not a rustc bug since the future correctly needs to borrow `&mut self` (to perform the spawn inside of the future, rather than before it gets polled the first time)

Reply:
Ah yes, I see. Makes sense.